packages/client/src/execution/vmexecution.ts

```diff
@@ -196,13 +197,19 @@ export class VMExecution extends Execution {
   }
 
   async setupVerkleVM() {
+    let stateManager: StatelessVerkleStateManager | StatefulVerkleStateManager
     if (this.verkleVM !== undefined) {
       return
     }
-    this.config.logger.info(`Setting up verkleVM`)
-    const stateManager = new StatelessVerkleStateManager({
-      common: this.config.execCommon,
-    })
+    if (this.config.statelessVerkle) {
+      this.config.logger.info(`Setting up verkleVM for stateless verkle execution`)
+      stateManager = new StatelessVerkleStateManager({
+        common: this.config.execCommon,
+      })
+    } else {
```

Comment:
Would it be best, instead of defaulting to stateful, to require users to explicitly mention either  `this.config.statelessVerkle` or `this.config.statefulVerkle`, and to throw if none of these are met?

Reply:
Good point, will fix.